inference-engine/tests/functional/plugin/shared/include/behavior/ov_exec_network.hpp

```diff
@@ -51,21 +166,217 @@ TEST_P(OVExecNetwork, precisionsAsInOriginalIR) {
+    ASSERT_NE(importedExecNet.output(0).get_node(), importedExecNet.output("concat").get_node());
+}
+
+TEST_P(OVExecNetwork, readFromV10IR) {
+    // Skip test according to plugin specific disabledTestPatterns() (if any)
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    std::string model = R"V0G0N(
+<net name="Network" version="10">
+    <layers>
+        <layer name="in1" type="Parameter" id="0" version="opset8">
+            <data element_type="f32" shape="1,3,22,22"/>
+            <output>
+                <port id="0" precision="FP32" names="data">
+                    <dim>1</dim>
+                    <dim>3</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                </port>
+            </output>
+        </layer>
+        <layer name="round" id="1" type="Round" version="opset8">
+            <data mode="half_to_even"/>
+            <input>
+                <port id="1" precision="FP32">
+                    <dim>1</dim>
+                    <dim>3</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                </port>
+            </input>
+            <output>
+                <port id="2" precision="FP32" names="r">
+                    <dim>1</dim>
+                    <dim>3</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                </port>
+            </output>
+        </layer>
+        <layer name="output" type="Result" id="2" version="opset8">
+            <input>
+                <port id="0" precision="FP32">
+                    <dim>1</dim>
+                    <dim>3</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                </port>
+            </input>
+        </layer>
+    </layers>
+    <edges>
+        <edge from-layer="0" from-port="0" to-layer="1" to-port="1"/>
+        <edge from-layer="1" from-port="2" to-layer="2" to-port="0"/>
+    </edges>
+</net>
+)V0G0N";
+    function = ie->read_model(model, InferenceEngine::Blob::Ptr());
+    ov::runtime::ExecutableNetwork execNet;
+    ASSERT_EQ(function->inputs().size(), 1);
+    ASSERT_EQ(function->outputs().size(), 1);
+    ASSERT_NO_THROW(function->input("in1"));
+    ASSERT_NO_THROW(function->output("round"));
+    ASSERT_NO_THROW(execNet = ie->compile_model(function, targetDevice, configuration));
+    ASSERT_EQ(execNet.inputs().size(), 1);
+    ASSERT_EQ(execNet.outputs().size(), 1);
+    ASSERT_NO_THROW(execNet.input("in1"));
+    ASSERT_NO_THROW(execNet.output("round"));
+
+    std::stringstream strm;
+    execNet.export_model(strm);
+
+    ov::runtime::ExecutableNetwork importedExecNet = ie->import_model(strm, targetDevice, configuration);
+    ASSERT_EQ(importedExecNet.inputs().size(), 1);
+    ASSERT_EQ(importedExecNet.outputs().size(), 1);
+    ASSERT_NO_THROW(importedExecNet.input("in1"));
+    ASSERT_NO_THROW(importedExecNet.output("round"));
```

Comment:
can we also check that precision are imported correctly? i.e. with CNNNetwork's conversion is applied
but need to change original precisions to f16